lib/faraday/sunset.rb

```diff
@@ -46,18 +47,63 @@ def report_deprecated_usage(env, datetime)
 
     def send_warning!(warning)
       warned = false
-      if @active_support
-        ActiveSupport::Deprecation.warn(warning)
-        warned = true
+
+      if @active_support == :auto
+        warned = report_active_support(warned, warning)
+      elsif @active_support == true
+        warned = report_active_support!(warned, warning)
       end
+
       if @logger && @logger.respond_to?(:warn)
         @logger.warn(warning)
         warned = true
       end
+
+      if @rollbar == :auto
+        warned = report_rollbar(warned, warning)
+      elsif @rollbar == true
+        warned = report_rollbar!(warned, warning)
+      end
+
       unless warned
-        raise NoOutputForWarning, "Pass active_support: true, or logger: ::Logger.new when registering middleware"
+        raise NoOutputForWarning, "Pass active_support: (true|false|:auto), rollbar: (true|false|:auto), or logger: ::Logger.new when registering middleware"
+      end
+    end
+
+    private
+
+    def report_rollbar!(warned, warning)
+      Rollbar.warning(warning)
+      # return true to set :warned
+      true
+    end
+
+    def report_active_support!(warned, warning)
+      ActiveSupport::Deprecation.warn(warning)
+      # return true to set :warned
+      true
+    end
+
+    # :auto methods
+    # do not raise errors if gems are missing
+    def report_rollbar(warned, warning)
+      begin
```

Comment:
#DidYouKnow: You can skip the `bein` here, and just `rescue` off the `def`.

Reply:
I did! It clearly slipped my mind here though.